### PR TITLE
Proper handling of removal and update of pokemon on the map

### PR DESCRIPTION
**Pull request by s7092910**
Changed the way that MarkerRefreshController behaves so it doesn't require another thread to be running constantly.

### CODE REVIEW

File: app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java [hunk 1]
```diff
@@ -505,14 +510,21 @@ public void onFetch(Bitmap bitmap) {
...
-                                    Marker marker = mGoogleMap.addMarker(new MarkerOptions()
+                                    final Marker marker = mGoogleMap.addMarker(new MarkerOptions()
                                             .position(new LatLng(poke.getLatitude(), poke.getLongitude()))
                                             .title(PokemonIdUtils.getLocalePokemonName(getContext(), poke.getPokemonId().name()))
                                             .icon(bitmapDescriptor)
                                             .zIndex(MapHelper.LAYER_POKEMONS)
                                             .anchor(0.5f, 0.5f));
                                     //adding pokemons to list to be removed on next search
-                                    markerList.put(poke.getSpawnPointId(), new PokemonMarkerExtended(poke, marker));
+                                    getActivity().runOnUiThread(new Runnable() {
```
Comment (rohanjain1) [inline app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java]: no need to run this separately on ui thread. we're already on the ui thread.

File: app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java [hunk 2]
```diff
@@ -18,38 +23,27 @@
...
-                    /**
-                     * Talking to:
-                     * {@link com.omkarmoghe.pokemap.views.map.MapWrapperFragment,}
-                     */
-                    EventBus.getDefault().post(new MarkerUpdateEvent());
+        Handler.Callback callback = new Handler.Callback() {
+            @Override
+            public boolean handleMessage(Message message) {
+                Log.d(TAG, "handleMessage: ");
+                if(message.obj instanceof PokemonMarkerExtended){
```
Comment (rohanjain1) [inline app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java]: setup and switch for a message.what, we may add other events to this later on.

File: app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java (hunk 1 above)
Comment (s7092910) [inline app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java]: Had to add that as sometimes for some reason it wasn't running on the UI thread which caused the messages to not be posted.

File: app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java [hunk 3]
```diff
@@ -64,44 +62,55 @@ public static MarkerRefreshController getInstance() {
...
+        if(mTimer != null){
+            mTimer.cancel();
+        }
+
+        long time = markerData.getCatchablePokemon().getExpirationTimestampMs() - System.currentTimeMillis();
+        final MarkerUpdate event = new MarkerUpdate(markerData);
+        mTimer = new CountDownTimer(time, DEFAULT_UPDATE_INTERVAL) {
+            @Override
+            public void onTick(long l) {
+                EventBus.getDefault().post(event);
```
Comment (rohanjain1) [inline app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java]: why do we need the event to be posted every second?

Comment (rohanjain1) [inline app/src/main/java/com/omkarmoghe/pokemap/controllers/MarkerRefreshController.java]: If timer ticks are needed for updating the snippet on marker, only run the timer tick for markers with visible infoWindows.

**Pull request merged**